lib/opFns.js

```diff
@@ -813,7 +826,9 @@ function checkCallMemCost (runState, callOptions, localOpts) {
   if (!callOptions.gasLimit) {
     callOptions.gasLimit = runState.gasLeft
   }
+}
 
+function checkOutOfGas (runState, callOptions) {
```

Comment:
Moving checkOutOfGas logic into a new function helps reduce the number of responsibilities for`checkCallMemCost`. This will help with maintainability, and implementation of EIP150 (so that the gas required for a call can be calculated separate from  `memCost` and `OOG Error` logic).